### PR TITLE
Support decimal as partition column

### PR DESCRIPTION
**Pull request by rui-mo**
Support decimal as partition column.

### CODE REVIEW

File: velox/connectors/hive/HiveConnector.cpp [hunk 1]
```diff
@@ -400,6 +400,45 @@ velox::variant convertFromString(const std::optional<std::string>& value) {
...
 }
 
+// A quick implementation to convert string as int128.
+int128_t atoint128_t(const char *s) {
+  const char *p = s;
+  int128_t val = 0;
+
+  if (*p == '-' || *p == '+') {
+    p++;
+  }
```
Comment (jinchengchenghh) [inline velox/connectors/hive/HiveConnector.cpp:410]: This implement does not consider overflow, is it possible overflow? I will have a PR to cast varchar to decimal. https://github.com/oap-project/velox/pull/169/files#diff-c6d2f99b68772f908f2f3e8b979b3c2894f790f20af68684e7b0400bed969389R222 Can you use this one after it merge?

Comment (jinchengchenghh) [reply velox/connectors/hive/HiveConnector.cpp:410]: This implement origins from velox https://github.com/facebookincubator/velox/blob/main/velox/type/Conversions.h#L89

Comment (rui-mo) [reply velox/connectors/hive/HiveConnector.cpp:410]: Sure. I'll take a look, thanks.

Comment (rui-mo) [reply velox/connectors/hive/HiveConnector.cpp]: @jinchengchenghh Updated. Can you help review again?

**Review by jinchengchenghh — APPROVED** [review]

**Pull request merged**